server/src/main/scala/org/http4s/server/middleware/authentication/DigestAuth.scala

```diff
@@ -35,11 +35,22 @@ import scala.concurrent.duration._
   */
 object DigestAuth {
 
+  type PlainAuthenticationStore[F[_], A] = String => F[Option[(A, String)]]
+  sealed trait AuthenticationStore[F[_], A]
+
   /** A function mapping username to a user object and password, or
     * None if no user exists.  Requires that the server can recover
     * the password in clear text, which is _strongly_ discouraged.
     */
-  type AuthenticationStore[F[_], A] = String => F[Option[(A, String)]]
+  final case class PlainTextAuthenticationStore[F[_], A](func: String => F[Option[(A, String)]])
+      extends AuthenticationStore[F, A]
```

Comment:
Annoying, but if we ever need to change the `case class` signature it is very difficult to retain bincompat across all 3 scala versions.

What if we do like:
```scala
trait AuthenticationStore[F[_], A] {
  def get(username: String): F[Option[(A, String)]]
}

sealed trait PlainTextAuthenticationStore[F[_], A] extends AuthenticationStore[F, A]
object PlainTextAuthenticationStore {
  def apply[F[_], A](func: String => F[Option[(A, String)]]) = ???
}

// etc.
```

Reply:
Ah, yeah, that's a good point. Will update.